src/backend/aspen/api/views/samples.py

```diff
@@ -1,27 +1,156 @@
+    DataType,
+    Entity,
+    GisaidAccession,
+    GisaidAccessionWorkflow,
+    Sample,
+    UploadedPathogenGenome,
+    User,
+)
 
 router = APIRouter()
 
+GISAID_REJECTION_TIME = datetime.timedelta(days=4)
+
 
 @router.get("/")
 async def list_samples(
     request: Request,
     db: AsyncSession = Depends(get_db),
     settings: Settings = Depends(get_settings),
     user: User = Depends(get_auth_user),
-) -> bool:
-    return False
+) -> SamplesResponseSchema:
+
+    cansee_groups_private_identifiers: Set[int] = {
+        cansee.owner_group_id
+        for cansee in user.group.can_see
+        if cansee.data_type == DataType.PRIVATE_IDENTIFIERS
+    }
+
+    # load the samples.
+    all_samples_query = sa.select(Sample).options(  # type: ignore
+        selectinload(Sample.uploaded_pathogen_genome),
+        selectinload(Sample.submitting_group),
+        selectinload(Sample.uploaded_by),
+    )
+    user_visible_samples_query = authz_samples_cansee(all_samples_query, None, user)
+    user_visible_samples_result = await db.execute(user_visible_samples_query)
+    user_visible_samples: List[Sample] = (
+        user_visible_samples_result.unique().scalars().all()
+    )
+
+    # Map gisaid accession workflows to samples.
+    # This should all go away when we change how GISAID ISL's are retrieved for samples.
+    SampleGisaidTuple = NamedTuple(
+        "SampleGisaidTuple",
+        [
+            ("sample", Sample),
+            ("uploaded_pathogen_genome", Optional[UploadedPathogenGenome]),
+            ("gisaid_accession_workflow", Optional[GisaidAccessionWorkflow]),
+            ("gisaid_accession", Optional[GisaidAccession]),
+        ],
+    )
+    sample_gisaid_tuple_map: Dict[int, SampleGisaidTuple] = dict()
+    genome_id_to_sample_id: Dict[int, int] = dict()
+    for sample in user_visible_samples:
+        if sample.uploaded_pathogen_genome is not None:
+            sample_gisaid_tuple_map[sample.id] = SampleGisaidTuple(
+                sample, sample.uploaded_pathogen_genome, None, None
+            )
+            genome_id_to_sample_id[
+                sample.uploaded_pathogen_genome.entity_id
+            ] = sample.id
+        else:
+            sample_gisaid_tuple_map[sample.id] = SampleGisaidTuple(
+                sample, None, None, None
+            )
+
+    gisaid_accession_workflows_inputs_query = (
+        sa.select(GisaidAccessionWorkflow)  # type: ignore
+        .join(GisaidAccessionWorkflow.inputs)  # type: ignore
+        .filter(Entity.id.in_(genome_id_to_sample_id.keys()))
+        .options(
+            selectinload(GisaidAccessionWorkflow.inputs),
+        )
+    )
+    gisaid_accession_workflows_with_inputs_response = await db.execute(
+        gisaid_accession_workflows_inputs_query
+    )
+    gisaid_accession_workflows_with_inputs: List[
+        GisaidAccessionWorkflow
+    ] = gisaid_accession_workflows_with_inputs_response.scalars().all()
+
+    # get around circular references
```

Comment:
Is it possible to run the above query with an alias instead of breaking this block into separate queries?
https://docs.sqlalchemy.org/en/14/orm/query.html#sqlalchemy.orm.aliased